awsutil/awsutil.go

```diff
@@ -29,6 +29,22 @@ func (e ECSTaskMeta) TaskID() string {
 	return split[len(split)-1]
 }
 
+func (e ECSTaskMeta) Region() string {
+	// Use the region from Task metadata if missing from environment (for EC2).
+	// https://github.com/aws/containers-roadmap/issues/337
+	for _, envKey := range []string{"AWS_REGION", "AWS_DEFAULT_REGION"} {
+		if region := os.Getenv(envKey); region != "" {
+			return region
+		}
+	}
```

Comment:
It feels a bit weird to me that this method on `ECSTaskMeta` is reaching out to env vars. I could see that be confusing in the future "Why is it returning this region when the metadata json response has _that_ region".

I think a function like:

```go
// Region returns the current AWS region by reading
// region environment variables and falling back to the
// region set in the task metadata.
func Region(meta ECSTaskMeta) string {}
```

Makes more sense to me but it's just an opinion. :) Should wait and see with Iryna thinks too.

Reply:
Makes sense to me! 

As I'm looking at this again, I'm wondering if I can rely on the SDK to do the environment variable discovery for us.

Then we could have a function that returns a config object for any AWS clients:

```
// ClientConfig returns a prepared config object for AWS clients.
// It pulls the region from the task metadata if not found in the environment.
func ClientConfig(meta ECSTaskMeta) *aws.Config {}
```

Reply:
Following my comment above, I added an `awsutil.NewSession` function, which relies on the AWS SDK to find the region from standard places, and falls back on Task meta if no region was found (also, allows us other session setup, like the user-agent handler).